test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpHandler.java

```diff
@@ -264,7 +275,100 @@ public void handle(final HttpExchange exchange) throws IOException {
+                                final String notFoundPart = Strings.format("""
+                                    --%s
+                                    Content-Type: application/http
+                                    Content-ID: %s
+
+                                    HTTP/1.1 404 The specified blob does not exist.
+                                    x-ms-error-code: BlobNotFound
+                                    x-ms-request-id: %s
+                                    x-ms-version: 2018-11-09
+                                    Content-Length: %d
+                                    Content-Type: application/xml
+
+                                    %s
+                                    """, responseBoundary, contentId, requestId, notFoundBody.length(), notFoundBody)
+                                    .replaceAll("\n", "\r\n");
+                                response.append(notFoundPart);
+                            }
+
+                            // Clear the state
+                            toDelete = null;
+                            contentId = null;
+                            requestId = null;
+                        } else if (Regex.simpleMatch("x-ms-client-request-id: *", line)) {
+                            if (requestId != null) {
+                                throw new IllegalStateException("Got multiple request IDs in a single request?");
+                            }
+                            requestId = line.split("\\s")[1];
+                        } else if (Regex.simpleMatch("Content-ID: *", line)) {
+                            if (contentId != null) {
+                                throw new IllegalStateException("Got multiple content IDs in a single request?");
+                            }
+                            contentId = line.split("\\s")[1];
+                        } else if (Regex.simpleMatch("DELETE /" + container + "/*", line)) {
+                            String blobName = RestUtils.decodeComponent(line.split("(\\s|\\?)")[1]);
+                            if (toDelete != null) {
+                                throw new IllegalStateException("Got multiple deletes in a single request?");
+                            }
+                            toDelete = blobName;
+                        }
+                    }
+                    response.append("--").append(responseBoundary).append("--\r\n0\r\n");
+                    // Send the response
+                    exchange.getResponseHeaders().add("Content-Type", "multipart/mixed; boundary=" + responseBoundary);
+                    exchange.sendResponseHeaders(RestStatus.ACCEPTED.getStatus(), response.length());
+                    logger.debug("--> Sending response:\n{}", response);
+                    try (OutputStream responseBody = exchange.getResponseBody()) {
+                        responseBody.write(response.toString().getBytes(StandardCharsets.UTF_8));
+                    }
```

Comment:
Nit: we don't actively close responseBody in other branch. I think it's closed in the finally block. So maybe it's better to keep it consistent?

Reply:
Fixed in. aa7ecfb7f768694b083ffc992015394123fda93e